src/main/java/frc/robot/subsystems/endEffector/TOFSensor.java

```diff
@@ -0,0 +1,82 @@
+package frc.robot.subsystems.endEffector;
+
+import com.playingwithfusion.TimeOfFlight;
+import com.playingwithfusion.TimeOfFlight.RangingMode;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class TOFSensor {
+  public double m_distance_EMA;
+  public double m_dist_SDEV_sq;
+  public String mode = "Short";
+
+  int m_rangeX0;
+  int m_rangeY0;
+  int m_rangeX1;
+  int m_rangeY1;
+
+  private TimeOfFlight sensor;
+
+  public TOFSensor(int CAN_ID) {
+    sensor = new TimeOfFlight(CAN_ID);
+    RoI(0, 0, 15, 15);
+    sensor.setRangingMode(RangingMode.Short, 0.24); // Will need to test. must be between 24-1000ms
+    m_distance_EMA = sensor.getRange();
+    m_dist_SDEV_sq = 0;
+
+  }
+
+  public TimeOfFlight.Status getStatus() {
+    return sensor.getStatus();
+  }
+  
+  public void RoI(int rangeX0, int rangeY0, int rangeX1, int rangeY1) {
+    m_rangeX0 = rangeX0;
+    m_rangeY0 = rangeY0;
+    m_rangeX1 = rangeX1;
+    m_rangeY1 = rangeY1;
+
+    sensor.setRangeOfInterest(m_rangeX0, m_rangeY0, m_rangeX1, m_rangeY1);
+    SmartDashboard.putNumber("rangeX0", m_rangeX0);
+    SmartDashboard.putNumber("rangeY0", m_rangeY0);
+    SmartDashboard.putNumber("rangeX1", m_rangeX1);
+    SmartDashboard.putNumber("rangeY1", m_rangeY1);
+  }
+
+  public double getRange() {
+    return sensor.getRange();
+  }
+
+  public void robotPeriodic() {
+
+    // /*for (int topLeftRow = 0; topLeftRow < 12; topLeftRow++) {
```

Comment:
Since we saved this code in the card, it should be okay to remove it and other commented code.